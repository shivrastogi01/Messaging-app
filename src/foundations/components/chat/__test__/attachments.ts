import exampleGiphy from "./exampleGiphy";

export const link = {
  url: "https://pubnub.com",
  provider: {
    name: "PubNub",
    url: "https://pubnub.com"
  },
  title:
    "Realtime Communication APIs for Chat, Notifications, Geolocation, and Collaboration | PubNub",
  description:
    "Leverage the power of hosted realtime APIs to build and augment connected experiences including in-app chat, mapping, device control and more on a massive scale.",
  image:
    "https://images.ctfassets.net/3prze68gbwl1/3roUbsKJAysxW6gUUxczrY/89eeaa8940ccf2ed10b348adaca99b64/pubnub-powered-apps.png",
  icon: "https://www.pubnub.com/images/favicon.png"
} as const;

export const image = {
  source:
    "https://images.ctfassets.net/3prze68gbwl1/7Mtx0mNP5byeuqrTr8gBCS/0d4ca318dc54edc7a17cfa9bf9987ecd/pubnub-chat-sdk-messaging-api.png"
};

export const gif = {
  title: "Giphy Logo",
  gif: exampleGiphy
};
