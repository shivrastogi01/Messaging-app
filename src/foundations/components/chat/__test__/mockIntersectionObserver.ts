export const mockIntersectionObserver = () => {
  class IntersectionObserver {
    observe = () => null;
    unobserve = () => null;
  }
  Object.defineProperty(window, "IntersectionObserver", {
    writable: true,
    configurable: true,
    value: IntersectionObserver
  });
  Object.defineProperty(global, "IntersectionObserver", {
    writable: true,
    configurable: true,
    value: IntersectionObserver
  });
};
