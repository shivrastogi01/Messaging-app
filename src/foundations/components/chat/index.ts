export * from "./text-message/text-message";
export * from "./image-message/image-message";
export * from "./link-message/link-message";
export * from "./giphy-message/giphy-message";
export * from "./avatar/avatar";
