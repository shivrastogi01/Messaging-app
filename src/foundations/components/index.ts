export * from "./presentation";
export * from "./layout";
