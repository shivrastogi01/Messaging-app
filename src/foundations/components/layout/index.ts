export * from "./styled-box/styled-box";
export * from "./scroll-view/scroll-view";
export * from "./flex-row/flex-row";
export * from "./flex-column/flex-column";
export * from "./modal/modal";
export * from "./drawer/drawer";
export * from "./list-item/list-item";
export * from "./dropdown/dropdown";
