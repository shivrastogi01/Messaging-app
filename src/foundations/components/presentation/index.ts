export * from "./heading/heading";
export * from "./label/label";
export * from "./title/title";
export * from "./icon/icon";
export * from "./button/button";
export * from "./input/input";
export * from "./textarea/textarea";
