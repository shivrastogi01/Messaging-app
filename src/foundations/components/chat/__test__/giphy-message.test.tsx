import React from "react";
import { GiphyMessage } from "foundations/components/chat";
import renderWithTheme from "main/testing/render-with-theme";
import { gif } from "./attachments";
import { mockIntersectionObserver } from "./mockIntersectionObserver";

beforeAll(() => {
  mockIntersectionObserver();
});

describe("GiphyMessage", () => {
  test("example gif matches the snapshot", () => {
    const { container } = renderWithTheme(
      <GiphyMessage
        title={gif.title}
        url={gif.gif.url}
        author={gif.gif.user.username}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("includes the title", () => {
    const { getByText } = renderWithTheme(
      <GiphyMessage
        title={gif.title}
        url={gif.gif.url}
        author={gif.gif.user.username}
      />
    );
    expect(getByText(gif.title)).toBeTruthy();
  });

  it("credits the author", () => {
    const { getByText } = renderWithTheme(
      <GiphyMessage
        title={gif.title}
        url={gif.gif.url}
        author={gif.gif.user.username}
      />
    );
    expect(getByText(gif.gif.username, { exact: false })).toBeTruthy();
  });

  it("provides GIPHY attribution", () => {
    const { getByAltText } = renderWithTheme(
      <GiphyMessage
        title={gif.title}
        url={gif.gif.url}
        author={gif.gif.user.username}
      />
    );
    expect(getByAltText("powered by GIPHY")).toBeTruthy();
  });
});
