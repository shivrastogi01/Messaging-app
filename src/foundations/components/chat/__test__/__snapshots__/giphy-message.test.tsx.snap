// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GiphyMessage example gif matches the snapshot 1`] = `
.c0 {
  background: #FFFFFF;
  border-radius: 10px;
  box-shadow: 0px 6px 10px rgba(103,19,176,0.2);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow: hidden;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 15px;
  font-weight: 300;
  height: 36px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0px 16px;
}

.c2 {
  color: #8E2DE2;
  font-weight: 500;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c3 {
  height: 16px;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div>
      <a
        class="c2"
        href="https://giphy.com/gifs/thisisgiphy-3o6gbbuLW76jkt8vIc"
        rel="noopener noreferrer"
        target="_blank"
      >
        Giphy Logo
      </a>
       by @originals
    </div>
    <img
      alt="powered by GIPHY"
      class="c3"
      src="attribution.png"
    />
  </div>
</div>
`;
